starter-code/src/AddNewFood.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from "react";
+
+class AddNewFood extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      name: "",
+      calories: "",
+      image: "",
+      quantity: ""
+    };
+  }
+
+  genericOnChange(event) {
+    const { value, name } = event.target;
+
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.foodSubmit(this.state);
+
+    this.setState({
+      name: "",
+      calories: "",
+      image: "",
+      quantity: ""
+    });
+  }
+
+  render() {
+    return (
+      <section className="AddFood">
+        <h2>Add Food</h2>
+        <form onSubmit={event => this.handleSubmit(event)}>
+          <label>
+            Name:
+            <input
+              onChange={event => this.genericOnChange(event)}
+              value={this.state.name}
+              type="text"
+              name="name"
+            />
+          </label>
+
+          <label>
+            Calories:
+            <input
+              onChange={event => this.genericOnChange(event)}
+              value={this.state.title}
+              type="number"
+              name="calories"
+            />
+          </label>
+
+          <label>
+            Image:
+            <input
+              onChange={event => this.genericOnChange(event)}
+              value={this.state.image}
+              type="url"
+              name="image"
+            />
+          </label>
+
+          <label>
+            Quantity:
+            <input
+              onChange={event => this.genericOnChange(event)}
+              value={this.state.quantity}
+              type="number"
+              name="quantity"
+            />
+          </label>
+
+          <button onClick={() => this.addFood}>Add Food</button>
```

Comment:
Here you could have used a ternary condition to hide the form `add a food`. 
```
      {addForm ? <button onClick={()=>this.addFormMethod()}>Add new food</button> : <FormFood addNewFood={newFood=>this.addNewFood(newFood)}/>}
```

with the two following functions to make this condition working: 
```
addFormMethod(){
    const {addForm} = this.state;
    this.setState ({addForm : !addForm})
  }

  addNewFood(newFood){
    const{foods, addForm} = this.state
    foods.unshift(newFood);
    this.setState({foods})
    this.setState({addForm : !addForm})
  }
```
and this in the state 
```
 this.state = {
      foods : foods,
      addForm : true, 
}
```